universal-application-tool-0.0.1/app/views/BaseHtmlView.java

```diff
@@ -21,17 +21,18 @@ protected Content htmlContent(DomContent... domContents) {
   }
 
   protected Tag textField(String fieldName, String labelText) {
-    return label(text(labelText), input().withType("text").withName(fieldName))
+    return label(text(labelText), input().withType("text").withName(fieldName).withId(fieldName))
```

Comment:
Out of curiosity, why are these IDs necessary now?

Reply:
Never mind, answered my own question - they are needed in the test

Reply:
Issue (blocking): `<input>` names and element IDs have different behavior and we should not encourage them defaulting them to the same value. It's common for multiple elements on the page or even within the same `<form>` (as is the case for checkboxes) to have the same name, but no two elements on the page should have the same ID.

Suggestion: have a second version of this method that allows specifying the ID.